kernel/thread.c

```diff
@@ -1167,14 +1167,36 @@ int k_thread_runtime_stats_all_get(k_thread_runtime_stats_t *stats)
 
 		stats->execution_cycles += tmp_stats.execution_cycles;
 		stats->total_cycles     += tmp_stats.total_cycles;
+#ifdef CONFIG_THREAD_RUNTIME_STATS_BETWEEN_COLLECTIONS
+		stats->execution_cycles_delta +=
+			tmp_stats.execution_cycles_delta;
+		stats->total_cycles_delta     +=
+			tmp_stats.total_cycles_delta;
+#endif
 #ifdef CONFIG_SCHED_THREAD_USAGE_ANALYSIS
 		stats->current_cycles   += tmp_stats.current_cycles;
 		stats->peak_cycles      += tmp_stats.peak_cycles;
 		stats->average_cycles   += tmp_stats.average_cycles;
 #endif
 		stats->idle_cycles      += tmp_stats.idle_cycles;
+#ifdef CONFIG_THREAD_RUNTIME_STATS_BETWEEN_COLLECTIONS
+		stats->idle_cycles_delta +=
+			tmp_stats.idle_cycles_delta;
+#endif
 	}
 #endif
 
 	return 0;
 }
+
+void k_thread_runtime_stats_all_cleanup(void)
```

Comment:
New API functions need docs.  In particular ones introducing new features.

Also not really loving the name.  "Clean up" implies that there is some garbage or something that needs to be tidied, and generally is used for steps that have to be done at some point.  This is closer to, maybe, "checkpoint": you're tagging a specific moment in the statistics data you want to use later.